torch/csrc/jit/type.cpp

```diff
@@ -85,4 +86,24 @@ ListTypePtr ListType::ofFloats() {
   return value;
 }
 
+// using the rules from python_arg_parser FunctionParameter::check
+// tensor must be 0 dim, and if the dest is an int the source must be integral type
+at::optional<std::string> checkScalarToNum(const TypePtr& source, const TypePtr& dest) {
```

Comment:
I am not sure this function is actually needed. It is only used in shape_propagation where we shouldn't be throwing errors anyway.